topi/python/topi/arm_cpu/conv2d.py

```diff
@@ -333,54 +332,11 @@ def _decl_winograd(cfg, data, kernel, strides, padding, dilation, layout, out_dt
     assert KH == 3 and KW == 3 and HPAD == 1 and WPAD == 1 and HSTR == 1 and WSTR == 1
```

Comment:
We still check kernel size=3？

Reply:
In fact, HPAD / WPAD shouldn't be checked too.

Reply:
@kevinthesun , @FrozenGene ,
* HPAD,WPAD asserts removed but assert KW == HW indtead (in all places)
* {KW, KH} == {3,3} assert removed but assert (kernel_size > 1) (in winograd_utils.py)

Reply:
I think the point here is that when should we use winograd. In autotvm tutorials, it will try to use winograd if no exception is thrown. If we make this change, all conv2d in resnet will use winograd implementation. Will it be faster than current implementation in which only 3*3 kernel size conv2d use winograd?

Reply:
@kevinthesun ,

1. In **one/same** we guard winograd matrix related (in)validity, already done [here](
https://github.com/dmlc/tvm/pull/3553/files#diff-f2ee9af361d4463abafcdb50b657529bR108)
  Taking it out from there means to maintain it in multiple unrelated places (each conv2d files).

2. For other (in)valid things (padding, stride, dilation etc) pure conv related things is good where they are already now (separate dedicated conv2d.py files for targeted arches: arm_cpu, mali, cuda, etc).

I would keep it simple with this PR, intention for now is to _only_ replace the winograd matrices, let's not touch functionalities or other actual limitations too much.

I'll keep in mind the suggestions here, next PR will be about enhancement around cfg.space (especialy arm/mali), then will be good time to revisit limitations.